src/tracing_tool_output.ml

```diff
@@ -113,33 +113,46 @@ module Serve = struct
   ;;
 end
 
+type events_output_format =
+  | Sexp
+  | Binio
+
+type events_writer =
+  { format : events_output_format
+  ; writer : Writer.t
+  ; output_callstack : bool
+  ; callstack_compression_state : Callstack_compression.t
+  }
+
 type t =
   { serve : Serve.t
-  ; output : [ `Fuchsia of string | `Sexp of string ]
+  ; fxt_path : string
+  ; events_path : string option
+  ; output_callstack : bool
   }
 
-let store_path = function
-  | `Fuchsia store_path | `Sexp store_path -> store_path
-;;
-
 let param =
-  let%map_open.Command store_path =
+  let%map_open.Command fxt_path =
     let default = "trace.fxt" in
     flag
       "output"
       (optional_with_default default string)
       ~aliases:[ "o" ]
-      ~doc:[%string "FILE Where to output the trace. (default: '%{default}')"]
+      ~doc:
+        [%string
+          "FILE Fuchsia file to output the fuchsia trace to. (default: '%{default}')"]
+  and events_path =
```

Comment:
Instead of adding a new argument here, what do you think about making it a function of the file extension given to `store_path`? e.g. if the file ends with `.sexp` or `.binio`, use your logic; else, do what it currently does.

Reply:
We could do this if we think it's important. The way the code is structured right now makes it a bit difficult as the code that reconstructs the call stacks is quite tightly integrated with the code that writes the fucshia file, and these would then need to be separated.

Reply:
In test, there's a way to replace the fuchsia file writing with something that logs each entry. Can you use that same interface to replace it with a "null writer" that just drops the fuchsia file entirely?